### PR TITLE
fix(core): update resourse data/ add possibility to update by querypa…

### PR DESCRIPTION
**Pull request by grankovsky**
update resourse data/ add possibility to update by queryparams/ add config to select params for update

### CODE REVIEW

**Review by Bozonych — APPROVED** [review]

File: packages/controllers/features/resource-data/resource-data.service.ts [hunk 1]
```diff
@@ -54,4 +75,19 @@ export class ResourceDataService<T extends IId = any> {
         this.data$.next(cloneDeep(this.stable));
         return this.data$;
     }
+
+    private shouldFilterParams(params: any): boolean {
```
Comment (denisogolubev) [inline packages/controllers/features/resource-data/resource-data.service.ts:79]: any?

File: packages/controllers/features/resource-data/resource-data.service.ts [hunk 2]
```diff
@@ -54,4 +75,19 @@ export class ResourceDataService<T extends IId = any> {
...
+        if (this.config?.updateEntityChangeParams) {
```
Comment (denisogolubev) [inline packages/controllers/features/resource-data/resource-data.service.ts:80]: 3 times access to the `this.config?.updateEntityChangeParams`, could be re-written to (use de-structurization):
```ts
const {updateEntityChangeParams} = this.config || {};
if (updateEntityChangeParams) {
    const param = params[updateEntityChangeParams[0]];
    const hasIdChanged: boolean = param !== this.previousId;
    this.previousId = param;
    return hasIdChanged;
}
return true;
```

File: packages/controllers/features/resource-data/resource-data.service.ts [hunk 3]
```diff
@@ -54,4 +75,19 @@ export class ResourceDataService<T extends IId = any> {
...
+            const hasIdChanged: boolean = params[this.config?.updateEntityChangeParams[0]] !== this.previousId;
+            this.previousId = params[this.config?.updateEntityChangeParams[0]];
+            return hasIdChanged;
+        }
+        return true;
+    }
+}
+
+interface DataResourceOptions{
```
Comment (denisogolubev) [inline packages/controllers/features/resource-data/resource-data.service.ts:89]: Interfaces should be in the `model` files

File: packages/controllers/features/resource-data/resource-data.service.ts [hunk 4]
```diff
@@ -27,15 +32,31 @@ export class ResourceDataService<T extends IId = any> {
...
+                        this.data$.next(data);
+                        this.stable = cloneDeep(data);
+                        return this.data$.pipe(shareReplay(1));
+                    }),
+                );
             }),
         );
     }
 
+    private hasRelevantParamChanges(currentParams: Params, relevantParams: string[]): boolean {
```
Comment (denisogolubev) [inline packages/controllers/features/resource-data/resource-data.service.ts:50]: You can use `some` iterator for this case:
```ts
relevantParams?.length && relevantParams.some((key: string) => {
    this.previousParams = {...currentParams};
    return this.shouldFilterParams(currentParams) && currentParams[key] !== this.previousParams[key];
});
```

But, you can keep it as it is. Just an alternative way to write this logic.

File: packages/controllers/features/resource-data/resource-data.service.ts (hunk 1 above)
Comment (grankovsky) [reply packages/controllers/features/resource-data/resource-data.service.ts]: fixed

File: packages/controllers/features/resource-data/resource-data.service.ts (hunk 2 above)
Comment (grankovsky) [reply packages/controllers/features/resource-data/resource-data.service.ts]: fixed

File: packages/controllers/features/resource-data/resource-data.service.ts (hunk 3 above)
Comment (grankovsky) [reply packages/controllers/features/resource-data/resource-data.service.ts]: replased

File: packages/controllers/features/resource-data/resource-data.service.ts (hunk 4 above)
Comment (grankovsky) [reply packages/controllers/features/resource-data/resource-data.service.ts:51]: let it as it is

**Review by denisogolubev — APPROVED** [review]

**Review by zverbeta — APPROVED** [review]

**Pull request merged**